src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -103,6 +104,39 @@ protected IMultipleTankHandler getOutputTank() {
         return controller.getOutputFluidInventory();
     }
 
+    @Override
+    protected boolean canWorkWithInputs() {
+        MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+        if (controller instanceof RecipeMapMultiblockController) {
+            RecipeMapMultiblockController distinctController = (RecipeMapMultiblockController) controller;
+
+            if (distinctController.canBeDistinct() && distinctController.isDistinct()) {
+                boolean canWork = false;
```

Comment:
Do we want to add a check here like
```java
if(!hasNotifiedInputs()){
        return false;
        }
```
to match the case in the `super` method and for short circuiting?

As if we do not have any notified inputs, then there is nothing for the multiblock to work with, and we have not checked if we have notified inputs yet.

Reply:
we clear the notified status after a recipe is run, so short circuiting it here would mean we run 1 recipe and stall till the input changes again